File: ares-fabric/src/main/java/dev/tigr/ares/fabric/impl/modules/movement/Baritone.java
package dev.tigr.ares.fabric.impl.modules.movement;

/*import baritone.api.BaritoneAPI;
import baritone.api.event.events.ChatEvent;*/

import dev.tigr.ares.core.feature.module.Category;
import dev.tigr.ares.core.feature.module.Module;
import dev.tigr.ares.core.setting.Setting;
import dev.tigr.ares.core.setting.settings.BooleanSetting;

/**
 * @author Tigermouthbear
 */
@Module.Info(name = "Baritone", description = "Allows you to change the settings for baritone", category = Category.MOVEMENT, enabled = true, visible = false, alwaysListening = true)
public class Baritone extends Module {
    private final Setting<Boolean> allowSprint = register(new BooleanSetting("Allow Sprint", true));
    private final Setting<Boolean> allowBreak = register(new BooleanSetting("Allow Break", true));
    private final Setting<Boolean> allowParkour = register(new BooleanSetting("Allow Parkour", true));
    private final Setting<Boolean> allowParkourPlace = register(new BooleanSetting("Allow Parkour Place", true));
    private final Setting<Boolean> allowInventory = register(new BooleanSetting("Manage Inventory", false));

    private final Setting<Boolean> allowDownward = register(new BooleanSetting("Allow Downward", true));
    private final Setting<Boolean> freeLook = register(new BooleanSetting("Freelook", true));
    private final Setting<Boolean> renderGoal = register(new BooleanSetting("Render Goal", true));
    private final Setting<Boolean> enterPortal = register(new BooleanSetting("Avoid Portals", false));

    /*public static <T> void executeCommand(String args) {
        boolean enabled = BaritoneAPI.getSettings().chatControl.value;
        BaritoneAPI.getSettings().chatControl.value = true;

        ChatEvent chatEvent = new ChatEvent(args);
        BaritoneAPI.getProvider().getPrimaryBaritone().getGameEventHandler().onSendChatMessage(chatEvent);
        if(!chatEvent.isCancelled())
            UTILS.printMessage(TextColor.RED + "Invalid Command!\n You can view a list possible commands at " + TextColor.BLUE + "https://github.com/cabaletta/baritone/blob/master/USAGE.md");

        BaritoneAPI.getSettings().chatControl.value = enabled;
    }

    @Override
    public void onTick() {
        BaritoneAPI.getSettings().allowSprint.value = allowSprint.getValue();
        BaritoneAPI.getSettings().allowBreak.value = allowBreak.getValue();
        BaritoneAPI.getSettings().allowParkour.value = allowParkour.getValue();
        BaritoneAPI.getSettings().allowParkourPlace.value = allowParkourPlace.getValue();
        BaritoneAPI.getSettings().allowInventory.value = allowInventory.getValue();
        BaritoneAPI.getSettings().allowDownward.value = allowDownward.getValue();
        BaritoneAPI.getSettings().freeLook.value = freeLook.getValue();
        BaritoneAPI.getSettings().renderGoal.value = renderGoal.getValue();
        BaritoneAPI.getSettings().enterPortal.value = enterPortal.getValue();
    }*/

    @Override
    public void onDisable() {
        setEnabled(true);
    }
}

File: src/main/kotlin/org/aresclient/ares/impl/instrument/modules/movement/Baritone.kt
package org.aresclient.ares.impl.instrument.modules.movement

import baritone.api.BaritoneAPI
import org.aresclient.ares.api.instruments.Module
import org.aresclient.ares.impl.util.BaritoneUtil

object Baritone: Module(Category.MOVEMENT, "Baritone", "Allows you to change the settings for baritone", defaults = Defaults().setAlwaysListening(true)) {
    private val allowSprint = settings.addBoolean("Allow Sprint", true).addListener(this::markDirty)
    private val allowBreak = settings.addBoolean("Allow Break", true).addListener(this::markDirty)
    private val allowParkour = settings.addBoolean("Allow Parkour", true).addListener(this::markDirty)
    private val allowParkourPlace = settings.addBoolean("Allow Parkour Place", true).addListener(this::markDirty)
    private val allowInventory = settings.addBoolean("Manage Inventory", false).addListener(this::markDirty)
    private val allowDownward = settings.addBoolean("Allow Downward", true).addListener(this::markDirty)
    private val freeLook = settings.addBoolean("Freelook", true).addListener(this::markDirty)
    private val renderGoal = settings.addBoolean("Render Goal", true).addListener(this::markDirty)
    private val enterPortal = settings.addBoolean("Avoid Portals", false).addListener(this::markDirty)

    private var dirty = true

    override fun onTick() {
        setEnabled(BaritoneUtil.isBaritonePresent()) // TODO: this probably isnt great to do every tick (need a way to override module toggling)
        if(!dirty) return

        BaritoneAPI.getSettings().allowSprint.value = allowSprint.value
        BaritoneAPI.getSettings().allowBreak.value = allowBreak.value
        BaritoneAPI.getSettings().allowParkour.value = allowParkour.value
        BaritoneAPI.getSettings().allowParkourPlace.value = allowParkourPlace.value
        BaritoneAPI.getSettings().allowInventory.value = allowInventory.value
        BaritoneAPI.getSettings().allowDownward.value = allowDownward.value
        BaritoneAPI.getSettings().freeLook.value = freeLook.value
        BaritoneAPI.getSettings().renderGoal.value = renderGoal.value
        BaritoneAPI.getSettings().enterPortal.value = enterPortal.value
    }

    private fun markDirty(value: Boolean) {
        dirty = true
    }
}
